var/spack/repos/builtin/packages/py-setuptools/package.py

```diff
@@ -69,15 +200,27 @@ def url_for_version(self, version):
     depends_on("python@2.7:2.8,3.5:", when="@44", type=("build", "run"))
     depends_on("python@2.7:2.8,3.4:", when="@:43", type=("build", "run"))
 
-    # Newer pip requires setuptools to be installed, before building
-    # setuptools. This issue was fixed or worked around in setuptools 54+
-    depends_on("py-pip@:18", when="@:53", type="build")
+    # https://github.com/pypa/setuptools/issues/3661
+    conflicts("python@3.12:", when="@:67")
 
-    # Uses HTMLParser.unescape
-    depends_on("python@:3.8", when="@:41.0", type=("build", "run"))
+    depends_on("py-pip", type="build")
 
-    # Uses collections.MutableMapping
-    depends_on("python@:3.9", when="@:40.4.2", type=("build", "run"))
```

Comment:
Also worth keeping this

Reply:
Did you mean this upperbound specifically, or the rest too?

Reply:
Just these two lines with the comment and upper bound